relay-server/src/services/processor.rs

```diff
@@ -192,6 +192,17 @@ impl ProcessingGroup {
             ))
         }
 
+        // Extract user report and user feedback.
+        let report_items = envelope.take_items_by(|item| {
+            matches!(item.ty(), &ItemType::UserReport | &ItemType::ClientReport)
```

Comment:
Client reports are very different from user reports: They carry outcomes, so like metrics they can be processed either completely separately or in a common envelope with other ungrouped items. 

The similarity in the name `ClientReport` and `UserReport` is merely coincidental.